frontend/src/providers/Provider.vue

```diff
@@ -0,0 +1,16 @@
+<template>
+  <GemProvider>
+    <slot></slot>
+  </GemProvider>
+</template>
```

Comment:
To confirm, each ProtoBuf service has a corresponding `Provider` that is defined here, right?

Reply:
Doesn't necessarily have to be a strict 1-1 mapping between ProtoBuf service and the Provider. Although I guess most of the time it would be? But yep, any additional providers would have to be added here

Reply:
Ok, thanks! I wasn't sure how having multiple ProtoBuf services would work since that would also indicate multiple clients.

Reply:
Hmm, you mean how it works with the proxy? 

My understanding is that if we initialise all the ProtoBuf services to send the request to the same host (localhost:10000), it'll proxy all the requests to the same backend server. So in the big picture, there's still only 3 components: client, server and proxy,

Reply:
> My understanding is that if we initialise all the ProtoBuf services to send the request to the same listener (localhost:10000), it'll proxy all the requests to the same backend server. So in the big picture, there's still only 3 components: 1 client, 1 server and 1 proxy,

I see, thanks! Seemed a bit weird to me coz for REST APIs we typically have a single API abstraction that sends all requests. Whereas here, we have `GemServicePromiseClient`, `UserServicePromiseClient`, etc. I guess the `client` terminology threw me off.